handlers/aws/utils.py

```diff
@@ -125,13 +125,87 @@ def wrapper(lambda_event: dict[str, Any], lambda_context: context_.Context) -> s
             if apm_client:
                 apm_client.capture_exception()
 
-            shared_logger.exception("exception raised", exc_info=e)
+            shared_logger.exception(
+                "exception raised",
+                exc_info=e,
+                extra={
+                    "summary": summarize_lambda_event(lambda_event, max_records=20),
+                },
+            )
 
             return f"exception raised: {e.__repr__()}"
 
     return wrapper
 
 
+def summarize_lambda_event(event: dict[str, Any], max_records: int = 10) -> dict[str, Any]:
+    """
+    Summarize the lambda event to include only the most relevant information.
+    """
+    summary: dict[str, Any] = {}
+
+    try:
+        records = event.get("Records", [])
+
+        for record in records:
+            event_source = record.get("eventSource", "unknown")
+
+            if event_source == "aws:sqs":
+                aws_sqs_summary = summary.get(
+                    "aws:sqs",
+                    # if the key does not exist, we initialize the summary
+                    {
+                        "total_records": 0,
+                        "records": [],
+                    },
+                )
+
+                # The body contains the SQS message payload which is the S3
+                # notification event encoded as a JSON string.
+                event = json_parser(record["body"])
```

Comment:
`event` variable already exists in the function parameters. I don't see it using elsewhere apart from being used to get the `records` so this wouldn't cause any issues, but are you overwriting the variable intentionally?

Reply:
Ouch, I did a bad thing here. Fixing it!

Reply:
`notifications` is probably a better name:

```python
                # The body contains the SQS message payload which is the S3
                # notification events encoded as a JSON string.
                notifications = json_parser(record["body"])


```